kitsune/groups/jinja2/groups/profile.html

```diff
@@ -18,10 +18,12 @@
         <a class="edit" href="{{ url('admin:groups_groupprofile_change', profile.id) }}">{{ _('Edit in admin') }}</a>
       {% endif %}
       <h1>{{ profile.group.name }}</h1>
-
-      {% if user_can_edit %}
-        <a class="edit" href="{{ url('groups.edit', profile.slug) }}">{{ _('Edit group profile') }}</a>
-      {% endif %}
+        {% if user.profile.is_staff %}
```

Comment:
This will not work when the user is not authenticated, since `user.profile` will be `None`. I just realized that putting `is_staff` -- which should be renamed as mentioned earlier -- on the profile means that we'll also have to constantly check if the user is authenticated or not, so now we'll have to use something like `user.is_authenticated and user.profile.is_staff` in nearly every place where we currently use `user.is_staff`.

It makes me wonder if we should abandon the idea of putting the `is_staff` on the profile, and instead go with the function idea I use within https://github.com/mozilla/kitsune/pull/5947. Within the file `kitsune/sumo/utils.py`, I created the function `is_staff_group_member`, which takes care of ensuring the user is defined and authenticated before making the DB call, and also caches the result on the user instance.

```python
def is_staff_group_member(user: User | None) -> bool:
    """
    Returns true only if the given user is a member of settings.STAFF_GROUP. If the user
    is authenticated, it also caches the result on the user instance so subsequent calls
    don't need to repeat the database query.
    """
    if not (user and user.is_authenticated):
        return False
    if not hasattr(user, "is_staff_group_member"):
        user.is_staff_group_member = user.groups.filter(name=settings.STAFF_GROUP).exists()
    return user.is_staff_group_member
```

So, with the function approach, we can replace `user.is_staff` with `is_staff_group_member(user)`, instead of `user.is_authenticated and user.profile.is_staff`.

Reply:
@escattone Could we meet in the middle and have the `Group` check live on as a `Profile` property and still use your helper function - but have your function check the `Profile` for `is_staff`? That way, we have:
1 - Explicit separation of the idea of authentication and group membership
2 - Consistent usage of the `Profile` as a place we store `User` properties (until we go to a custom `User` model)

Your function might combine concerns that should be separate - for example, a use case where I want to check if a non-authenticated `User` has group membership, won't this just return` False`? So we'd be using the longer `user.groups.filter(name=settings.STAFF_GROUP).exists()` more for those cases....

Thoughts?

Reply:
NOTE: We had a video discussion on what name to use other than `is_staff` and settled on `in_staff_group` for now.

Since any user instance that isn't an `AnonymousUser` instance is guaranteed to have `is_authenticated == True`, there really can't be any separation between authentication and the possibility of group membership. A user has to be authenticated -- in other words, not an `AnonymousUser` -- in order to be a member of one or more groups or have a profile. So we'd always have to check if the user is authenticated before trying `user.profile.in_staff_group`.

As you said, we could still meet in the middle, and assuming we make `in_staff_group` a `cached_property`, we could add the following to `kitsune/sumo/utils.py`:

```python
def in_staff_group(user: User | None) -> bool:
    """
    Returns true only if the given user is a member of settings.STAFF_GROUP.
    """
    return bool(user and user.is_authenticated and user.profile.is_staff_group_member)
```

and then also add a Jinja helper function within `kitsune/sumo/templatetags/jinja_helpers.py`:

```python
from kitsune.sumo.utils import in_staff_group
...
library.global_function(in_staff_group)
```

Reply:
I've implemented `in_staff_group`, added `cached_property` and added both the `utils.py` and `jinja_helpers.py` changes.